autoload/fern/internal/replacer.vim

```diff
@@ -97,16 +97,26 @@ function! s:BufWriteCmd() abort
     for Modifier in b:fern_replacer_modifiers
       let result = Modifier(result)
     endfor
-    let l:Resolve = b:fern_replacer_resolve
-    set nomodified
-    close
-    call Resolve(result)
   catch
     echohl ErrorMsg
     echo '[fern] Please fix the following error first to continue or cancel with ":q!"'
     echo printf('[fern] %s', substitute(v:exception, '^Vim(.*):', '', ''))
     echohl None
+    return
   endtry
+  set nomodified
+  let l:Resolve = b:fern_replacer_resolve
+  let winnr = winnr()
+  let bufnr = bufnr()
+  call timer_start(0, { -> s:post_write(l:Resolve, result, winnr, bufnr) })
+endfunction
+
+function! s:post_write(resolve, result, winnr, bufnr) abort
+  if winbufnr(a:winnr) == a:bufnr
+    execute a:winnr . 'wincmd c'
+  endif
```

Comment:
How about using win_execute instead?

Reply:
Use `win_execute` and `window-ID` instead `winnr`.